seed/views/v3/data_quality.py

```diff
@@ -0,0 +1,478 @@
+from django.http import JsonResponse, HttpResponse
+from rest_framework import viewsets, serializers, status
+from rest_framework.decorators import action
+from unidecode import unidecode
+
+from seed.data_importer.tasks import do_checks
+from seed.decorators import ajax_request_class
+from seed.lib.superperms.orgs.decorators import has_perm_class
+from seed.lib.superperms.orgs.models import (
+    Organization,
+)
+from seed.models.data_quality import (
+    Rule,
+    DataQualityCheck,
+)
+from seed.utils.api import api_endpoint_class
+from seed.utils.api_schema import AutoSchemaHelper
+from seed.utils.cache import get_cache_raw
+
+logger = get_task_logger(__name__)
+
+
+class RulesSubSerializer(serializers.Serializer):
+    field = serializers.CharField(max_length=100)
+    severity = serializers.CharField(max_length=100)
+
+
+class RulesSubSerializerB(serializers.Serializer):
+    field = serializers.CharField(max_length=100)
+    enabled = serializers.BooleanField()
+    data_type = serializers.CharField(max_length=100)
+    min = serializers.FloatField()
+    max = serializers.FloatField()
+    severity = serializers.CharField(max_length=100)
+    units = serializers.CharField(max_length=100)
+
+
+class RulesIntermediateSerializer(serializers.Serializer):
+    missing_matching_field = RulesSubSerializer(many=True)
+    missing_values = RulesSubSerializer(many=True)
+    in_range_checking = RulesSubSerializerB(many=True)
+
+
+class RulesSerializer(serializers.Serializer):
+    data_quality_rules = RulesIntermediateSerializer()
+
+
+def _get_js_rule_type(data_type):
+    """return the JS friendly data type name for the data data_quality rule
+
+    :param data_type: data data_quality rule data type as defined in data_quality.models
+    :returns: (string) JS data type name
+    """
+    return dict(Rule.DATA_TYPES).get(data_type)
+
+
+def _get_js_rule_severity(severity):
+    """return the JS friendly severity name for the data data_quality rule
+
+    :param severity: data data_quality rule severity as defined in data_quality.models
+    :returns: (string) JS severity name
+    """
+    return dict(Rule.SEVERITY).get(severity)
+
+
+def _get_rule_type_from_js(data_type):
+    """return the Rules TYPE from the JS friendly data type
+
+    :param data_type: 'string', 'number', 'date', or 'year'
+    :returns: int data type as defined in data_quality.models
+    """
+    d = {v: k for k, v in dict(Rule.DATA_TYPES).items()}
+    return d.get(data_type)
+
+
+def _get_severity_from_js(severity):
+    """return the Rules SEVERITY from the JS friendly severity
+
+    :param severity: 'error', or 'warning'
+    :returns: int severity as defined in data_quality.models
+    """
+    d = {v: k for k, v in dict(Rule.SEVERITY).items()}
+    return d.get(severity)
+
+
+class DataQualitySchema(AutoSchemaHelper):
+    def __init__(self):
+        super().__init__()
+
+        self.manual_fields = {
+            ('POST', 'create'): [
```

Comment:
Can anyone think of a case where (`HTTP method`, `ViewSet method/function name`) would not be enough to identify a ViewSet method?

Reply:
not that I can think of with this level of information. Can you give an example?

Reply:
I agree and think a ViewSet method has a unique pair of values for HTTP method and ViewSet function name. I was thinking that ViewSet function name would be enough, but I figured that having both couldn't hurt.